vendor/github.com/containers/common/pkg/auth/auth.go

```diff
@@ -0,0 +1,167 @@
+package auth
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/pkg/docker/config"
+	"github.com/containers/image/v5/types"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh/terminal"
+)
+
+func GetDefaultAuthFile() string {
+	return os.Getenv("REGISTRY_AUTH_FILE")
+}
+
+func CheckAuthFile(authfile string) error {
+	if authfile == "" {
+		return nil
+	}
+	if _, err := os.Stat(authfile); err != nil {
+		return errors.Wrapf(err, "error checking authfile path %s", authfile)
+	}
+	return nil
+}
+
+// GetCredsAndLogin login to the server with creds from Stdin or CLI
+func GetCredsAndLogin(ctx context.Context, systemContext *types.SystemContext, iopts *LoginOptions, server, userFromAuthFile, passFromAuthFile string) error {
```

Comment:
(Non-blocking: Not sure about the name, when this is not returning any credentials.)